### ISSUE TITLE
Incorrect orders updated

### ISSUE THREAD

**Issue by shachiarkatkar** [magento/magento2#19258]
### Preconditions (*)
1. Magento 2.2.5
2. Php 7.0

### Steps to reproduce (*)
1. Open order grid in two tabs.
2. Apply filter of order status in both tabs. Both tabs should show same orders. (ex. pending)
3. Select first order in first tab and change the status of order(ex. change to hold). 
4. Without refresh select same order in second tab and update status(ex. change to cancel)

### Expected result (*)
1. It should show exception message saying "Cant update order" or "No order to update" with id
![image](https://user-images.githubusercontent.com/2101412/48661022-92de4d80-ea91-11e8-9e54-9362d472ad81.png)

### Actual result (*)
1. All orders were updated to cancel status except the one updated in first tab.
![image](https://user-images.githubusercontent.com/2101412/48661008-5ad70a80-ea91-11e8-92f9-3cea6ae5c9f4.png)

**Comment by magento-engcom-team** [comment]
Hi @shachiarkatkar. Thank you for your report. 
To help us process this issue please make sure that you provided the following information:
- [ ] Summary of the issue
- [ ] Information on your environment
- [ ] Steps to reproduce
- [ ] Expected and actual results

Please make sure that the issue is reproducible on the vanilla Magento instance following **Steps to reproduce**. To deploy vanilla Magento instance on our environment, please, add a comment to the issue: 

`@magento-engcom-team give me $VERSION instance` 

where `$VERSION` is [version tags](https://github.com/magento/magento2/tags) (starting from 2.2.0+) or develop branches (for example: 2.3-develop). 
For more details, please, review the [Magento Contributor Assistant](https://devdocs.magento.com/guides/v2.2/contributor-guide/contributing.html#contributor-assist) documentation.

@shachiarkatkar do you confirm that you was able to reproduce the issue on vanilla Magento instance following steps to reproduce?
- [ ] yes
- [ ] no

**Comment by shachiarkatkar** [comment]
This issue replicated in  vanilla Magento instance

**Comment by shachiarkatkar** [comment]
@magento-engcom-team have checked on clean 2.2.5 and 2.3-dev branch got same error

**Comment by konarshankar07** [comment]
@shachiarkatkar
 I think the orders which are on hold status are not able to cancel that's the way Magento works. Hope this will help you.

**Comment by shachiarkatkar** [comment]
@konarshankar07 its not just about hold status but the other orders shouldn't be updated. Please check Following code in vendor/magento/module-ui/Component/MassAction/Filter.php:

 * Adds filters to collection using DataProvider filter results
     *
     * @param AbstractDb $collection
     * @return AbstractDb
     * @throws LocalizedException
     */
    public function getCollection(AbstractDb $collection)
    {
        $selected = $this->request->getParam(static::SELECTED_PARAM);
        $excluded = $this->request->getParam(static::EXCLUDED_PARAM);

        $isExcludedIdsValid = (is_array($excluded) && !empty($excluded));
        $isSelectedIdsValid = (is_array($selected) && !empty($selected));

        if ('false' !== $excluded) {
            if (!$isExcludedIdsValid && !$isSelectedIdsValid) {
                throw new LocalizedException(__('Please select item(s).'));
            }
        }
        /** @var \Magento\Customer\Model\ResourceModel\Customer\Collection $collection */
        $idsArray = $this->getFilterIds();
        if (!empty($idsArray)) {
            $collection->addFieldToFilter(
                $collection->getIdFieldName(),
                ['in' => $idsArray]
            );
        }
        return $collection;
    }

It returns whole collection because $idsArray is empty, if the order is already updated in other tab as it is not actually present in grid. Hence whole collection is affected. Please guide.

**Comment by konarshankar07** [comment]
@shachiarkatkar 
I tried to reproduce the issue but I'm not able to do it. I followed the following step to reproduce.
1)On the first tab, I change the order status to hold
2)On the second tab, I change the order status to unhold (without reloading the tab).
Its working fine in 2.2.5. Let me know if I missed any step to reproduce this issue.
Thanks

**Comment by shachiarkatkar** [comment]
Try to cancel the order in second tab

**Comment by ratnesh-kadam** [comment]
@magento-engcom-team 
Please follow following steps to reproduce the error:
1. Open order grid in two tabs.
2. Apply filter of order status in both tabs. Both tabs should show same orders.
3. Select first order in first tab and change the status of order. 
4. Without refresh select same order in second tab and update status
------------------------------------------
Expected result:
It should throw error that cant update order as it is already updated and not showing in that filtered view.
------------------------------------------
Actual Result:
All orders are updated except the one selected.

**Comment by magento-engcom-team** [comment]
Hi @engcom-backlog-andrii. Thank you for working on this issue. 
In order to make sure that issue has enough information and ready for development, please read and check the following instruction: :point_down:
- [x] 1. Verify that issue has all the required information. (**Preconditions**, **Steps to reproduce**, **Expected result**, **Actual result**).<details><summary>Details</summary>If the issue has a valid description, the label `Issue: Format is valid` will be added to the issue automatically. Please, edit issue description if needed, until label `Issue: Format is valid` appears.</details>

- [x] 2. Verify that issue has a meaningful description and provides enough information to reproduce the issue. If the report is valid, add `Issue: Clear Description` label to the issue by yourself.

- [x] 3. Add `Component: XXXXX` label(s) to the ticket, indicating the components it may be related to.
- [x] 4. Verify that the issue is reproducible on `2.3-develop` branch<details><summary>Details</summary>- Add the comment `@magento-engcom-team give me 2.3-develop instance` to deploy test instance on Magento infrastructure. <br/> - If the issue is reproducible on `2.3-develop` branch, please, add the label `Reproduced on 2.3.x`.<br/>- If the issue is not reproducible, add your comment that issue is not reproducible and close the issue and **_stop verification process here_**!
</details>

- [x] 5. Verify that the issue is reproducible on `2.2-develop` branch. <details><summary>Details</summary>- Add the comment `@magento-engcom-team give me 2.2-develop instance` to deploy test instance on Magento infrastructure. <br/>- If the issue is reproducible on `2.2-develop` branch, please add the label `Reproduced on 2.2.x`</details>

- [x] 6. Add label `Issue: Confirmed` once verification is complete. 

- [x] 7. Make sure that automatic system confirms that report has been added to the backlog.

**Comment by magento-engcom-team** [comment]
@engcom-backlog-andrii Thank you for verifying the issue. Based on the provided information internal tickets `MAGETWO-96556`, `MAGETWO-96557` were created

**Comment by krnshah** [comment]
Hi @shachiarkatkar,

As per your given step i have tried to reproduce the issue but i cant reproduce.

-- Set filter to pending status in both the tab.
-- Select the order and change the order status of that order in first tab.
-- Select the same order in second tab after redirect to the order detail. **It will not giving any error msg**. But the order status will change as we select in first tab.

Please let me know if i am wrong.

**Comment by ghost** [comment]
Hi @krnshah. But we must got message after that. In second tab there is no message  "Can't update order" or some else after update status  as it is already updated in first tab and not showing in that filtered view.

**Comment by shikhamis11** [comment]
@magento-engcom-team give me 2.3-develop instance

**Comment by magento-engcom-team** [comment]
Hi @shikhamis11. Thank you for your request. I'm working on Magento 2.3-develop instance for you

**Comment by magento-engcom-team** [comment]
Hi @shikhamis11, here is your [Magento instance](https://i-19258-2-3-develop.instances.magento-community.engineering).
Admin access: [https://i-19258-2-3-develop.instances.magento-community.engineering/admin](https://i-19258-2-3-develop.instances.magento-community.engineering/admin)
Login: `admin` Password: `123123q`
Instance will be terminated in up to 3 hours.

**Comment by magento-engcom-team** [comment]
Hi @shachiarkatkar. Thank you for your report. 
The issue has been fixed in magento/magento2#20349 by @shikhamis11 in 2.3-develop branch
Related commit(s):
 - [56c1b3395cad0b34362c6df4bf5ca74a40d9a363](https://github.com/magento/magento2/commit/56c1b3395cad0b34362c6df4bf5ca74a40d9a363)
 - [d55f47d0d34031ab602702ec0b2fc518d5cc0b9c](https://github.com/magento/magento2/commit/d55f47d0d34031ab602702ec0b2fc518d5cc0b9c)

The fix will be available with the upcoming 2.3.1 release.

**Comment by irajneeshgupta** [comment]
@magento-engcom-team give me 2.2-develop instance

**Comment by magento-engcom-team** [comment]
Hi @irajneeshgupta. Thank you for your request. I'm working on Magento 2.2-develop instance for you

**Comment by magento-engcom-team** [comment]
Hi @irajneeshgupta, here is your [Magento instance](https://i-19258-2-2-develop.instances.magento-community.engineering).
Admin access: [https://i-19258-2-2-develop.instances.magento-community.engineering/admin](https://i-19258-2-2-develop.instances.magento-community.engineering/admin)
Login: `admin` Password: `123123q`
Instance will be terminated in up to 3 hours.
